src/main/java/org/hydev/mcpm/Controller.java

```diff
@@ -0,0 +1,80 @@
+package org.hydev.mcpm;
+
+import net.sourceforge.argparse4j.inf.ArgumentParserException;
+import org.hydev.mcpm.client.injector.*;
+
+import java.util.function.Consumer;
+
+import static java.lang.String.format;
+import static org.hydev.mcpm.utils.Sugar.sub;
+import static org.hydev.mcpm.utils.Sugar.subFrom;
+
+
+/**
+ * Controller of the program
+ *
+ * @author Azalea (https://github.com/hykilpikonna)
+ * @since 2022-10-30
+ */
+public class Controller
+{
+    private final LoadBoundary loader;
+    private final UnloadBoundary unloader;
+    private final ReloadBoundary reloader;
+
+    public Controller()
+    {
+        var pl = new PluginLoader();
+        loader = pl;
+        unloader = pl;
+        reloader = pl;
+    }
+
+    /**
+     * Run a command
+     *
+     * @param args Command-line arguments (not including the command name/path)
+     * @param log Callback logger
+     */
+    public void runCommand(String[] args, Consumer<String> log)
+    {
+        if (args.length == 0) return;
+
+        var cmd = args[0].toLowerCase();
```

Comment:
Oops, I didn't notice that subcommands are supported 🥲 

We can change this when we add another command, or we can also keep it this way so that the help messages are shorter